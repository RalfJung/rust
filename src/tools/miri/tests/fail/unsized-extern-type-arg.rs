#![feature(extern_types)]
#![feature(unsized_fn_params)]

extern {
    pub type E;
}

fn test(_e: E) {}

pub fn calltest(e: Box<E>) {
    test(*e) //~ERROR: does not have a dynamically computable size
}

fn main() {
    let b = Box::new(0u32);
    calltest(unsafe { std::mem::transmute(b)} );
}
