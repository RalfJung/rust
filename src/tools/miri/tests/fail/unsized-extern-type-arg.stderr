error: function parameter does not have a dynamically computable size
  --> $DIR/unsized-extern-type-arg.rs:LL:CC
   |
LL |     test(*e)
   |     ^^^^^^^^
   |
   = note: `E` contains an `extern type`, which is not allowed in function parameters

error: aborting due to previous error

